components/style/properties/gecko.mako.rs

```diff
@@ -1447,6 +1284,186 @@ fn static_assert() {
     }
 </%def>
 
+fn set_image(image: Image, mut geckoimage: &mut structs::nsStyleImage) {
+    match image {
+        Image::Gradient(gradient) => {
+            set_gradient(gradient, &mut geckoimage)
+        },
+        Image::Url(..) => {
```

Comment:
There should be a third argument, `with_url`, that turns on this branch. (`Image::Url if with_url => ..`). The `if shorthand == "background"` above needs to set with_url.